### DISCUSSION TITLE
Collecting output from subworkflow

### DISCUSSION THREAD

**Discussion by bts76-pitt** [nextflow-io/nextflow#4782] [Q&A] [answered]
Hello!

I'm currently developing a pipeline which processes multiple FASTQ files using a subworkflow and piped processes, which places the output in a specific directory. Then afterwards, I'm attempting to run a process that works on the entire directory of output data from the subworkflow. Each step of the pipeline works fine on it's own, but chaining them together has been an issue. I've tried the "collect" operation, as well as defining a dummy process to delay the second process from running until the output folder has been populated by the subworkflow, but the final process executes and fails before all of the samples are processed by the subworkflow. Are there any suggestions how best to chain the subworkflow and the process together?

Thanks in advance!

**Comment by mribeirodantas** [comment, marked as answer]
Why are you trying to make your subworkflow read from the `publishDir` directory instead of the work directory? Publishing files work asynch. You should rather make your subworkflow read output channesls from the processes/subworkflows that you're interested in.

**Reply by bts76-pitt** [reply]
Thanks, I ended up incorporating the final process into the subworkflow by having the last process in the subworkflow emit a dummy boolean. Using the "collect" operation I was able to avoid the asynchronous running of the final process while also avoiding the issue you mentioned - so I don't need to read from the publishDir because everything I need is in the working directory. I appreciate your response!
